tests/common-tests.h

```diff
@@ -172,6 +172,66 @@ std::ostream& operator<<(std::ostream& ostr, const std::vector<MyOptionalString>
     return ostr;
 }
 
+// The next classes are used for DDL testing
+template<typename T>
+struct DdlTester {
+    typedef T val_type;
+    DdlTester(soci::db_type type, T val, bool expectNull)
```

Comment:
Do I understand correctly that `val` is not used if `expectNull` is true? I.e. both these parameters could be replaced with `std::optional<T>`?

Reply:
val is still pushed to the database before getting the result.
This is to test the handling of the value NaN which is interprested by sqlite as null. Others databases does not handle NaN the same way.

Reply:
I think it would make more sense to add a separate test checking for NaN to null conversion if we really want it (as I said, I'm still not sure if we actually do) rather than reusing this -- apparently unrelated -- test for this, at the cost of a lot of extra complications.